ledger/sandbox-on-x/src/main/scala/com/daml/ledger/sandbox/ReadServiceWithFeedSubscriber.scala

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml
+
+import ledger.api.health.{HealthStatus, Healthy}
+import ledger.configuration.{Configuration, LedgerId, LedgerInitialConditions, LedgerTimeModel}
+import ledger.offset.Offset
+import ledger.participant.state.v2.{ReadService, Update}
+import lf.data.Time.Timestamp
+import logging.{ContextualizedLogger, LoggingContext}
+import akka.NotUsed
+import akka.stream.Materializer
+import akka.stream.scaladsl.Source
+import org.reactivestreams.Subscriber
+import scala.util.chaining._
+
+trait ReadServiceWithFeedSubscriber extends ReadService {
+  def subscriber: Subscriber[(Offset, Update)]
```

Comment:
why do we need to make the subscriber (or directly the Sink if we can use MergeHub as well) part of the ReadService with inheritance here?
For me it would be more natural just to have a Source dependency here, and let factories build that source for this.

Reply:
Just for encapsulation (i.e. not passing Akka stream API objects around in the SandboxOnXRunner). Do you see a problem with this?

Reply:
But now I see a problem that we introduce a false directed dependency between the read -> write service. Ok, will extract

Reply:
Extracted.